packages/run-protocol/src/vaultFactory/vaultManager.js

```diff
@@ -254,79 +254,94 @@ const helperBehavior = {
-    // ratio of debt to collateral is below the liquidationMargin, and if so,
-    // we liquidate it. We use ceilDivide to round up because ratios above
-    // this will be liquidated.
-    const quoteRatioPlusMargin = makeRatioFromAmounts(
-      ceilDivideBy(getAmountOut(quote), liquidationMargin),
-      getAmountIn(quote),
-    );
-    trace('quote', quote, quoteRatioPlusMargin);
-
-    state.outstandingQuote = null;
-
-    // Liquidate the head of the queue
-    const [next] =
-      prioritizedVaults.entriesPrioritizedGTE(quoteRatioPlusMargin);
-    await (next ? facets.helper.liquidateAndRemove(next) : null);
-    trace('price check liq', next && next[0]);
-
-    // eslint-disable-next-line consistent-return
-    return facets.helper.reschedulePriceCheck();
+  processLiquidations: async ({ state, facets }) => {
+    const { prioritizedVaults, priceAuthority } = state;
+    const govParams = state.factoryPowers.getGovernedParams();
+
+    async function* liquidations() {
+      while (true) {
+        const highestDebtRatio = prioritizedVaults.highestRatio();
+        if (!highestDebtRatio) {
+          return;
+        }
+        const liquidationMargin = govParams.getLiquidationMargin();
+
+        // ask to be alerted when the price level falls enough that the vault
+        // with the highest debt to collateral ratio will no longer be valued at the
+        // liquidationMargin above its debt.
+        const triggerPoint = ceilMultiplyBy(
+          highestDebtRatio.numerator, // debt
+          liquidationMargin,
+        );
+        state.outstandingQuote = E(priceAuthority).mutableQuoteWhenLT(
+          highestDebtRatio.denominator, // collateral
+          triggerPoint,
+        );
+        trace('posted quote request', triggerPoint);
+
+        // The rest of this method will not happen until after a quote is received.
+        // This may not happen until much later, when themarket changes.
+        // eslint-disable-next-line no-await-in-loop
+        const quote = await E(state.outstandingQuote).getPromise();
+        state.outstandingQuote = null;
+        // When we receive a quote, we check whether the vault with the highest
+        // ratio of debt to collateral is below the liquidationMargin, and if so,
+        // we liquidate it. We use ceilDivide to round up because ratios above
+        // this will be liquidated.
+        const quoteRatioPlusMargin = makeRatioFromAmounts(
+          ceilDivideBy(getAmountOut(quote), liquidationMargin),
+          getAmountIn(quote),
+        );
+        trace('quote', quote, quoteRatioPlusMargin);
+
+        // Liquidate the head of the queue
+        const [next] =
+          prioritizedVaults.entriesPrioritizedGTE(quoteRatioPlusMargin);
+        if (next) {
+          yield next;
+        }
+      }
+    }
+    for await (const next of liquidations()) {
+      await facets.helper.liquidateAndRemove(next);
+      trace('price check liq', next && next[0]);
+    }
```

Comment:
seeing just this part it looks like once a liquidation is triggered by a drop, all vaults under water will be liquidated regardless of further price movement. I suggest a jsdoc comment on `liquidations()` fn def that it yields one vault per quote resolution. (basically repeating the "will not happen until after a quote is received" aspect )

Reply:
rename `liquidations` to `eventualLiquidations`